extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1631,8 +1636,9 @@ private void checkPendingCompletionTasks() throws ExecutionException, Interrupte
 
           // reset partitions offsets for this task group so that they will be re-read from metadata storage
           partitionGroups.get(groupId).replaceAll((partition, offset) -> NOT_SET);
-          sequenceTaskGroup.remove(generateSequenceName(groupId));
-
+          if (taskGroups.get(groupId) != null) {
+            sequenceTaskGroup.remove(generateSequenceName(groupId));
```

Comment:
What causes taskGroups to not have an entry for groupId? I wonder if there should be a warning log here

Reply:
The groupId might get removed from `taskGroups` in `checkTaskDuration()`, as described in https://github.com/druid-io/druid/issues/5900 . A warning log could be added.

Reply:
Looking at the code more closely, it seems in `checkTaskDuration()`, before removing `groupId` from `taskGroups`, it's removed from `sequenceTaskGroup` as follows:
```sequenceTaskGroup.remove(generateSequenceName(groupId));
      // remove this task group from the list of current task groups now that it has been handled
      taskGroups.remove(groupId);```
In `checkPendingCompletionTasks()` originally this call was causing NPE `sequenceTaskGroup.remove(generateSequenceName(groupId));`, so I am wondering if we  remove this call ? Or add a check like I am doing currently to be safe ?